README.md

```diff
@@ -1,2 +1,95 @@
+   // Events notifies of session creation after navigation, UA action, or requestSession.
+   // The session object is provided as part of this event.
+   if (evt.session.mode === 'immersive-vr') {
+      // set up app state for immersive vr, if that's what the app wants
+   } else {
+      // notify user that this app only works in immersive vr mode, if desired
+   }
+}
+```
+
+This is similar to the [WebVR displayactivate event](https://immersive-web.github.io/webvr/spec/1.1/#dom-window-onvrdisplayactivate) that popular libraries and frameworks incorporate alongide user gesture activated flows. The UA always remains in control on when immersive sessions are granted and can also revoke them.
+
+## Security considerations
+
+In immersive mode the Browser UI may not be rendered and content controls all display real estate. The UA must steward users giving them information and control over displayed content at all times.
+
+Spoofing is a potential risk during immersive navitagion. A malicious site could masquerade legitimate content, the browser UI or even the underling operating system interface without user noticing. Possible mitigaton mechanisms are:
+
+- Interstitial. When navigation occurs the UA can present the user a screen to evaluate the destination URL, an experience description and a button to continue or return to previous site.
+- Overlays. UAs control compositing and can always render information about the navigated page over content.
+- Modal dialog. User can invoke the browser UI at all times to interrupt or consult info about the current site. A modal interface (similar to Oculus or Steam home spaces) could be showed on demand with a reserved button or interaction that cannot be listened by content. To prevent spoofing, UAs could additionally display personal information that user can recognize and pages don't have access to (user generated doodles or environment choices)
+- Fall back to windowed mode. Some UAs might decide to retain a more traditional flow and always fallback to windowed / 2D mode on navigation.
+
+## Developer and library maintainers considerations.
+
+The `sessiongranted` event is low friction and agnostic. It doesn't tie or prescribe a specific browser UX allowing experimentation and innovation in the space. Some browsers might decide to offer a traditional windowed workflow that falls back to 2D, others a more immersive experience. Content creators and library maintainers have to consider just two flows: User and UA initiated immersive sessions via user gesture and sessiongranted event respectively.
+
+## Privacy considerations.
+
+### Exposing State About the Client or Previous Page
+
+Varying the behavior based on client state (e.g., is the UA displaying multiple tabs) or the state of the previous page (e.g., whether the previous page had an active immersive session) leaks state about the user and user’s behavior that is not otherwise available to applications.
```

Comment:
I'm not sure I see how anything new is exposed here beyond what a page would receive in 2D mode.  Please explain.

Reply:
I think you're right here. This was copied from the navigation explainer and I think the issue is if users can't control page navigation, their data could be exposed further

Reply:
@AlbertoElias Any ideas to expand / reword / clarify?